scrapers/distgit.py

```diff
@@ -0,0 +1,171 @@
+            del soup
+
+            log.debug('Creating nodes associated with commit {0} and push {1}'.format(
+                result['commit_id'], result['push_id']))
+            repo = DistGitRepo.get_or_create({
+                'namespace': namespace,
+                'name': result['module']
+            })[0]
+            branch_name = result['ref'].rsplit('/', 1)[1]
+            branch = DistGitBranch.get_or_create(
+                {'name': branch_name}, relationships=repo.branches)[0]
+            commit = DistGitCommit.get_or_create({
+                'id_': result['commit_id'],
+                'author_date': result['author_date'],
+                'commit_date': result['commit_date'],
+                'sha': result['sha'],
+                'log_message': result['log_message']
+            })[0]
+            push = DistGitPush.get_or_create({
+                'id_': result['push_id'],
+                'push_date': result['push_date'],
+                'push_ip': result['push_ip']
+            })[0]
+            bug = BugzillaBug.get_or_create({'id_': result['bugzilla_id']})[0]
+
+            log.debug('Creating the user nodes associated with commit {0} and push {1}'.format(
+                result['commit_id'], result['push_id']))
+            author = User.create_or_update({
+                'username': people['author_username'],
+                'email': people['author_email']
+            })[0]
+            committer = User.create_or_update({
+                'username': people['committer_username'],
+                'email': people['committer_email']
+            })[0]
+            pusher = User.get_or_create({
+                'username': result['pusher']
+            })[0]
+
+            log.debug('Creating the relationships associated with commit {0} and push {1}'.format(
+                result['commit_id'], result['push_id']))
+            author.distgit_authored_commits.connect(commit)
+            author.distgit_branches.connect(branch)
+            author.distgit_repos.connect(repo)
+
+            committer.distgit_committed_commits.connect(commit)
+            committer.distgit_branches.connect(branch)
+            committer.distgit_repos.connect(repo)
+
+            pusher.distgit_pushes.connect(push)
+            pusher.distgit_branches.connect(branch)
+            pusher.distgit_repos.connect(repo)
+
+            repo.contributors.connect(author)
+            repo.contributors.connect(committer)
+            repo.contributors.connect(pusher)
+            repo.commits.connect(commit)
+            repo.pushes.connect(push)
+            repo.branches.connect(branch)
+
+            branch.commits.connect(commit)
+            branch.pushes.connect(push)
+
+            push.owners.connect(pusher)
+            push.pushers.connect(pusher)
+            push.commits.connect(commit)
+            push.branches.connect(branch)
+            push.repos.connect(repo)
+
+            commit.owners.connect(author)
```

Comment:
Just one suggestion, if there are any attributes that can be null at times, you might want to adjust the position in which you define your relationships. For eg, in Koji, if there is no task_id for the build, then I wanted to skip the remaining loop, so I defined the relationships between the 'user' and 'builds' before I checked for a task_id. 

I don't know if something like this applies to this scraper :)

Reply:
Thanks for the reminder. I don't think there are any null values I'll come across but I'm running the scraper now and will let go all the way if time permits to ensure that is the case.